test/gtest/uct/test_event.cc

```diff
@@ -109,6 +110,10 @@ void test_uct_event_fd::test_recv_am(bool signaled)
     recv_buffer = (recv_desc_t *) malloc(sizeof(*recv_buffer) + sizeof(send_data));
     recv_buffer->length = 0; /* Initialize length to 0 */
 
+    /* give a chance to finish connection for some transports (ib/ud, tcp) */
+    m_e1->flush();
```

Comment:
will TCP return NO_RESOURCE if connection is not established yet?
If yes, you would fail here, because entity flush progresses while IN_PROGRESS only.
better use uct_test flush then

Reply:
yes, tested - it fails with the non-blocking code.
thank you! fixed

btw, it discovered the wrong implementation of uct_iface_flush() in TCP for non-blocking connect.